### PR TITLE
DidChangeTextDocumentParams parses elisp JSON

### PR DESCRIPTION
**Pull request by alanz**
Which uses "null" for []

### CODE REVIEW

File: src/Language/Haskell/LSP/TH/DataTypesJSON.hs [hunk 1]
```diff
@@ -18,6 +18,20 @@ import Data.Default
 
 -- ---------------------------------------------------------------------
 
+-- | This data type is used to host a FromJSON instance for the encoding used by
+-- elisp, where an empty list shows up as "null"
+data List a = List [a]
```
Comment (adamwalker) [inline src/Language/Haskell/LSP/TH/DataTypesJSON.hs]: Would a newtype be better?

**Comment by alanz** [comment]
It is for the xxxJSON instances. Would that work with a newtype?

If so it does not really matter either way.

**Comment by adamwalker** [comment]
I believe it would work. It doesn't really matter. 

The advantage of a newtype, I believe, is that it has no runtime overhead. List a would have exactly the same representation in memory as [a]. I don't think this is true when "data" is used.

**Comment by alanz** [comment]
Ok, I will change it.

**Pull request merged**